quint/src/cliCommands.ts

```diff
@@ -553,53 +547,60 @@ export async function runSimulator(prev: TypecheckedStage): Promise<CLIProcedure
   const mainEnd = prev.sourceMap.get(mainId)!.end!.index
   const result = compileAndRun(newIdGenerator(), mainText, mainStart, mainEnd, mainName, mainPath, options)
 
-  if (result.status !== 'error') {
-    if (verbosity.hasResults(verbosityLevel)) {
-      const elapsedMs = Date.now() - startMs
+  const elapsedMs = Date.now() - startMs
+
+  switch (result.outcome.status) {
```

Comment:
This is now case analysis, so there is one, and only one, branch for each outcome. This also ensures that we'll get static checks to enforce we handle new statuses if they are introduced.